lua/lspconfig/denols.lua

```diff
@@ -1,8 +1,42 @@
 local configs = require 'lspconfig/configs'
 local util = require 'lspconfig/util'
+local lsp = vim.lsp
 
 local server_name = "denols"
 
+local function buf_cache(bufnr)
+  local params = {}
+  params["referrer"] = { textDocument = { uri = vim.uri_from_bufnr(bufnr) } }
+  params["uris"] = {}
+  lsp.buf_request(bufnr, "deno/cache", params)
+end
+
+local function virtual_text_document_handler(uri, result)
+  if not result or #result ~= 1 then return nil end
+
+  local lines = vim.split(result[1].result, "\n")
+  local bufnr = vim.uri_to_bufnr(uri)
+
+  local current_buf = vim.api.nvim_buf_get_lines(bufnr, 0, -1, false)
+  if #current_buf ~= 0 then return nil end
+
+  vim.api.nvim_buf_set_lines(bufnr, 0, -1, nil, lines)
+  vim.api.nvim_buf_call(bufnr, function()
+    vim.bo.readonly = true
```

Comment:
Hmm, luacheck was failed. Is there the way to resolve it?

```
lua/lspconfig/denols.lua:25:5: setting read-only field bo.readonly of global vim
```

Reply:
I don't know actually, I believe this  should work. @tjdevries do you have any ideas about this?

To work around you can try `:help nvim_buf_set_option`

Reply:
`nvim_buf_set_option` works well. Thank you!

Reply:
I think this is probably the more "correct" way to do than buf_call, but it's weird luacheck gave you issues

Reply:
https://github.com/neovim/nvim-lspconfig/pull/746

Reply:
Thank you, I saw. I think we should probably prefer nvim_buf_set_option generally since it's more specific, but I'd be in favor of merging #746